File: gemini-service/src/main/java/studio/styx/erisbot/services/gemini/GeminiRequest.java
package studio.styx.erisbot.services.gemini;

import com.google.genai.Client;
import com.google.genai.types.GenerateContentConfig;
import com.google.genai.types.GenerateContentResponse;
import shared.utils.Env;

import java.util.Objects;

public class GeminiRequest {
    private final Client client = Client.builder()
            .apiKey(Objects.requireNonNull(Env.get("GEMINI_API_KEY")).toString())
            .build();

    public GenerateContentResponse request(String prompt) {
        return client.models.generateContent(
                        "gemini-2.5-flash",
                        prompt,
                        null);

    }

    public GenerateContentResponse request(String prompt, GenerateContentConfig config) {
        return client.models.generateContent(
                "gemini-2.5-flash",
                prompt,
                config
        );
    }
}

File: gemini-service/src/main/kotlin/services/gemini/GeminiRequest.kt
package services.gemini

import com.google.genai.Client
import com.google.genai.types.GenerateContentConfig
import com.google.genai.types.GenerateContentResponse
import shared.utils.Env

class GeminiRequest {
    private val client: Client = Client.builder()
        .apiKey(Env.get("GEMINI_API_KEY").toString())
        .build()

    fun request(prompt: String): GenerateContentResponse? {
        return client.models.generateContent(
            "gemini-2.5-flash",
            prompt,
            null
        )
    }

    fun request(prompt: String, config: GenerateContentConfig?): GenerateContentResponse? {
        return client.models.generateContent(
            "gemini-2.5-flash",
            prompt,
            config
        )
    }
}